src/components/SearchBar.tsx

```diff
@@ -101,11 +117,21 @@ export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
         let searchOptionsBtn = <button id="SearchOptionsBtn" onClick={this.onExpandButtonClick.bind(this)}><i className="fa fa-angle-double-down fa-2x"></i></button>;
         let thisObj = this;
         let checkboxes: JSX.Element[] = [];
+        let cancelButton: JSX.Element = null;
+
+        _.each(this.props.categories, function (c: string) {
+            let checked = this.isCategorySelected(c);
+            checkboxes.push(this.createCheckbox(c, "CategoryCheckbox", checked, this.onCategoriesChanged.bind(thisObj)));
+        }.bind(this));
```

Comment:
I'm sure `thisObj` reference here isn't necessary anymore (and the overall `thisObj` can be removed). Be careful when this is removed, other code below should be verified to ensure no breakage.